hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/NSSummaryTaskDbEventHandler.java

```diff
@@ -194,4 +195,24 @@ protected void handleDeleteDirEvent(OmDirectoryInfo directoryInfo,
     nsSummary.removeChildDir(objectId);
     nsSummaryMap.put(parentObjectId, nsSummary);
   }
+
+  protected boolean flushAndCommitNSToDB(Map<Long, NSSummary> nsSummaryMap) {
+    try {
+      writeNSSummariesToDB(nsSummaryMap);
+    } catch (IOException e) {
+      LOG.error("Unable to write Namespace Summary data in Recon DB.", e);
+      return false;
+    }
+    return true;
+  }
+
+  protected boolean checkAndCallFlushToDB(
+      Map<Long, NSSummary> nsSummaryMap) {
+    // if map contains more than entries, flush to DB and clear the map
+    if (null != nsSummaryMap && nsSummaryMap.size() ==
```

Comment:
Check should be nsSummaryMap.size() >= NSSUMMARY_FLUSH_TO_DB_MAX_THRESHOLD

Reply:
@sumitagrawl - this is fixed. Pls re-review